lib/pbench/cli/agent/commands/tools/clear.py

```diff
@@ -0,0 +1,190 @@
+"""
+pbench-clear-tools
+
+This script will remove tools that have been registered.  If no options are
+used, then all tools from the "default" tool group are removed.  Specifying
+a tool name and/or remote host will limit the scope of the removal.
+
+
+"""
+import shutil
+import sys
+import os
+import pathlib
+
+import click
+
+from pbench.agent.utils import setup_logging
+from pbench.cli.agent import CliContext, pass_cli_context
+from pbench.cli.agent.commands.tools.base import ToolCommand
+from pbench.cli.agent.options import common_options
+
+
+class ClearTools(ToolCommand):
+    """Clear registered tools"""
+
+    def __init__(self, context):
+        super().__init__(context)
+
+        self.logger = setup_logging(logfile=self.pbench_log)
+
+    def execute(self):
+        errors = 0
+
+        self.tool_group_dir = self.tool_group_dir(self.context.group)
+        if not self.tool_group_dir.exists() or self.tool_group_dir is None:
+            return 1
+
+        try:
+            remotes = self.context.remote.split(",")
+        except Exception:
+            # We were not given any remotes on the command line, build the list
+            # from the tools group directory.
+            remotes = self.remote(self.tool_group_dir)
+
+        for remote in remotes:
+            tg_dir_r = self.tool_group_dir / remote
+            if not tg_dir_r.exists():
+                self.logger.warn(
+                    'The given remote host, "%s", is not a directory in' " %s.",
+                    remote,
+                    self.tool_group_dir,
+                )
+                continue
+
+            if self.context.name:
+                names = self.context.name
+            else:
+                # Discovery all the tools registered for this remote
+                names = self.tools(tg_dir_r)
+
+            for name in names:
+                status = self._clear_tools(name, remote)
+                if status != 0:
+                    errors += 1
+
+            tool_files = [p.name for p in tg_dir_r.iterdir() if p.name == "__label__"]
```

Comment:
This is compiling a list of all the files named `__label__` and deleting if there's only one; but the original code was `$(/bin/ls ${tools_group_dir}/${remote})` to ensure that `__label__` was the _only_ file in the directory.

I think you need to drop the `if p.name == "__label__"` to be equivalent.